diff --git a/tests/koka/.metadata/compiled-files b/tests/koka/.metadata/compiled-files
index 7a55800..8f48207 100644
--- a/tests/koka/.metadata/compiled-files
+++ b/tests/koka/.metadata/compiled-files
@@ -1,5 +1,4 @@
 /Users/gaisseml/dev/effect-jit/rpyeffect-asm/tests/koka/hello.kk
-/Users/gaisseml/dev/effect-jit/rpyeffect-asm/tests/koka/test.kk
 /Users/gaisseml/dev/effect-jit/rpyeffect-asm/tests/koka/test4.kk
 /Users/gaisseml/dev/effect-jit/rpyeffect-asm/tests/koka/test3.kk
 /Users/gaisseml/dev/effect-jit/rpyeffect-asm/tests/koka/test2.kk
diff --git a/tests/koka/.metadata/date b/tests/koka/.metadata/date
index 55648f2..db26090 100644
--- a/tests/koka/.metadata/date
+++ b/tests/koka/.metadata/date
@@ -1 +1 @@
-2024-02-22T10:04:05+00:00
+2024-02-22T11:10:04+00:00
